core/src/main/java/com/datastax/oss/driver/internal/core/graal/DefaultFeature.java

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.graal;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Optional;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeClassInitialization;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+/**
+ * A Graal feature which determines which checks for well-known config keys corresponding to
+ * instances to be instantiated via reflection, adding each to the current reflection config.
+ */
+@AutomaticFeature
+final class DefaultFeature implements Feature {
+
+  public Config buildConfig() {
+
+    /* Copied from DefaultDriverConfigLoader, including the DEFAULT_ROOT_PATH value there as the literal
+    "datastax-java-driver" below.  Can't reference the class directly here since that will introduce
+    a requirement to specify it as a build-time dependency (as well as the slf4j logger it creates
+    via static init */
+    return ConfigFactory.defaultOverrides()
+        .withFallback(ConfigFactory.defaultApplication())
+        .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader()))
+        .resolve()
+        .getConfig("datastax-java-driver");
+  }
+
+  public Optional<String> tryString(Config config, DefaultDriverOption option) {
+
+    try {
+      return Optional.of(config.getString(option.getPath()));
+    } catch (ConfigException.WrongType e) {
+      return Optional.empty();
+    } catch (ConfigException.Missing e) {
+      return Optional.empty();
+    }
+  }
+
+  public Optional<Class<?>> tryLoadClass(String configVal, String... defaultPackages) {
+
+    try {
+      if (configVal.contains(".")) {
+        return Optional.of(Class.forName(configVal));
+      } else {
+        for (String defaultPackage : defaultPackages) {
+          String qualifiedClassName = defaultPackage + "." + configVal;
+          try {
+            return Optional.of(Class.forName(qualifiedClassName));
+          } catch (ClassNotFoundException cnfe) {
+          }
+        }
+        return Optional.empty();
+      }
+    } catch (LinkageError | Exception e) {
+      return Optional.empty();
+    }
```

Comment:
We wind up re-implementing some of the bits here from Reflection.java but I don't see an obvious way around that.  The config is written to assume the presence of this logic when processing entries (the default for LBP is literally "DefaultLoadBalancingPolicy" under the assumption the package stuff will clean it up at instantiation time) so we're left with some redundancy.

It would be nice to consolidate this into a single impl used by both... but there are other pressing matters to work on at the moment.